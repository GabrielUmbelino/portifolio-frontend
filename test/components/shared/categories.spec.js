import { shallowMount } from '@vue/test-utils'
import Categories from '@/components/shared/categories'

describe('Project Stats', () => {
  it('Render categories with values', async () => {
    const wrapper = shallowMount(Categories, {
      propsData: { categories: [{ id: 1, description: 'Foo Bar' }] },
      mocks: {
        $t: () => 'Gabriel tomando um pau dos testes',
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test.each([[], null, undefined])(
    'Do not render when categories are %s',
    (categories) => {
      const wrapper = shallowMount(Categories, {
        propsData: { categories },
      })

      expect(wrapper.element).toMatchSnapshot()
    }
  )
})
