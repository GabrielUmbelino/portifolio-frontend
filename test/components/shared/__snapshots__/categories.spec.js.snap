// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Stats Do not render when categories are [] 1`] = `<!---->`;

exports[`Project Stats Do not render when categories are null 1`] = `<!---->`;

exports[`Project Stats Do not render when categories are undefined 1`] = `<!---->`;

exports[`Project Stats Render categories with values 1`] = `
<div
  class="categories"
>
  <h4
    class="subtitle"
  >
    Gabriel tomando um pau dos testes
  </h4>
  <span
    class="ant-tag tag"
  >
    <span>
      Foo Bar
    </span>
  </span>
</div>
`;
